cpp_utils/test/unittest/thread_pool/slot_thread_pool_test.cpp

```diff
@@ -82,6 +82,9 @@ TEST(slot_thread_pool_test, pool_one_thread_one_slot)
     // Wait for counter value to be greater than 0 (so 1 task is being executed)
     waiter.wait_greater_equal_than(test::N_EXECUTIONS_IN_TEST);
 
+    // Join threads before destroying waiter to avoid data race
+    thread_pool.disable();
```

Comment:
NIT: it is unlikely but this could still head to an error. It should use `blocking_disable`